Sources/Datadog/RUMMonitor.swift

```diff
@@ -84,11 +84,9 @@ public enum RUMUserActionType {
 }
 
 public enum RUMErrorSource {
+    case custom
     case source
-    case console
     case network
-    case agent
-    case logger
```

Comment:
We need `.logger` to be used in `RUMIntegrations` and I guess this is why we replace `RUMErrorSource` with `RUMDataSource` in `RUMCommand` (?). As mentioned earlier, this negatively impacts the data layers isolation. Can we come up with a different approach? 

One idea could be to introduce internal `enum _RUMErrorSource` extending the original one with te extra case and use it instead of `RUMErrorSource`:
```swift
public enum RUMErrorSource {
   case custom, source, network, webview

   internal var dd: _RUMErrorSource {
      switch self {
         case .custom: .custom
         case .source: .source
         case .network: .network
         case .webview: .webview
      }
   }
}

internal enum _RUMErrorSource {
   case custom, source, network, webview, logger
}
```

Another is to replace `enum` with `struct` (users don't need enumeration-semantic for `RUMErrorSource`), leverage different access modifiers for different options:
```swift
public struct RUMErrorSource {
   public static let custom = "custom"
   public static let source = "source"
   public static let network = "network"
   public static let webview = "webview"
   internal static let logger = "logger"
}
```
and finding out the way of ensuring safety in tests.

Reply:
i think there is no way to test it completely with a `struct`
i added `RUMInternalErrorSource` enum ✅